x-pack/solutions/search/plugins/search_indices/public/hooks/use_ingest_data.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { i18n } from '@kbn/i18n';
+import type { IHttpFetchError, ResponseErrorBody } from '@kbn/core/public';
+import type { InstallResponse } from '@kbn/sample-data-ingest/common';
+import { useKibana } from './use_kibana';
+
+import { QueryKeys } from '../constants';
+
+type ServerError = IHttpFetchError<ResponseErrorBody>;
+
+export function useIngestSampleData() {
+  const {
+    sampleDataIngest,
+    notifications: { toasts },
+  } = useKibana().services;
+  const queryClient = useQueryClient();
+
+  const { mutate: ingestSampleData, isLoading } = useMutation<InstallResponse, ServerError, void>(
+    [QueryKeys.IngestSampleData],
+    () => {
+      if (!sampleDataIngest) {
+        throw new Error('sampleDataIngest plugin not installed');
+      }
+
+      return sampleDataIngest?.install();
+    },
+    {
+      onSuccess: ({ indexName }) => {
```

Comment:
If we're not using the `indexName` we don't need to define it here.
```suggestion
      onSuccess: () => {
```